src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -564,7 +564,7 @@ private void CreateGrainInstance(string grainTypeName, ActivationData data, stri
             Type stateObjectType = grainTypeData.StateObjectType;
             lock (data)
             {
-                var grain = (Grain) Activator.CreateInstance(grainType);
+                var grain = (Grain) Runtime.Silo.CurrentSilo.DependencyResolver.GetService(grainType);
```

Comment:
would there always be DependencyResolver? How do we make sure that at least the default one is set?